### DISCUSSION TITLE
Cannot execute migration with optional field: "23502"

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#13421] [Q&A] [answered]
I want to add a column to an table but the migration is failing:

The model:

```
model Notification {
  id               Int              @id @default(autoincrement())
  movie            Movie?           @relation(fields: [movieId], references: [id])
  movieId          Int? 
  movieRating      MovieRating?     @relation(fields: [movieRatingId], references: [id])
  movieRatingId    Int? 
  user             User?            @relation(fields: [userId], references: [id])
  userId           Int? 
  followedUser     User?            @relation("FollowedUser", fields: [followedUserId], references: [id])
  followedUserId   Int? 
  action           String 
  value            String?
  watched          Boolean          @default(false)
} 
```
The error:

> Database error:
ERROR: column "value" of relation "Notification" contains null values
> DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("23502"), message: "column \"value\" of relation \"Notification\" contains null values", detail: None, hint: None, position: None, where_: None, schema: Some("public"), table: Some("Notification"), column: Some("value"), datatype: None, constraint: None, file: Some("tablecmds.c"), line: Some(5450), routine: Some("ATRewriteTable") }

Screengrab of the table:
![image](https://user-images.githubusercontent.com/779921/169381635-a1606a25-1939-4043-aac9-f6a063ecef2f.png)

I understand that if the `value` column that I want to add is required that this would show an error, but the `String?` makes it's optional right?

**Comment by alucardu** [comment, marked as answer]
Took a closer look in my migrations and I added the value column in 2 steps. First I added it without the optional ? but since my local db was empty there was no issue. Then I made it optional, but on production it was the first migration (adding the value without optional) that failed.

My first migration was: AlterTable ALTER TABLE "Notification" ADD COLUMN     "value" TEXT NOT NULL;

After that I created another migration: ALTER TABLE "Notification" ALTER COLUMN "value" DROP NOT NULL;

So this is what caused the error.

I resolved it by rolling back the first migration:  npx prisma migrate resolve --rolled-back "20220515174738_added_value_field_to_notificaion"

Then I changed the first migration file: AlterTable ALTER TABLE "Notification" ADD COLUMN     "value" TEXT; and I removed the second migration.

Now I can deploy the migrations again :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
